iree/hal/cts/command_queue_test.cc

```diff
@@ -26,7 +26,17 @@ namespace {
 using ::iree::testing::status::IsOkAndHolds;
 using ::testing::Eq;
 
-class CommandQueueTest : public CtsTestBase {};
+class CommandQueueTest : public CtsTestBase {
+ protected:
+  virtual void SetUp() {
+    CtsTestBase::SetUp();
+
+    // Disabled on Vulkan until tests pass with timeline semaphore emulation.
+    if (driver_->name() == "vulkan") {
+      GTEST_SKIP();
+    }
```

Comment:
Skip before setup?

```suggestion
class CommandQueueTest : public CtsTestBase {
 protected:
  virtual void SetUp() {
    // Disabled on Vulkan until tests pass with timeline semaphore emulation.
    if (driver_->name() == "vulkan") {
      GTEST_SKIP();
    }
    CtsTestBase::SetUp();
```


Otherwise it tries to set up vulkan anyway, it looks like

```
[22:00:53 UTC] /tmpfs/src/github/iree/build/iree/hal/cts/iree_hal_cts_command_queue_test
[==========] Running 20 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 20 tests from AllDrivers/CommandQueueTest
[ RUN      ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/dylib
I ../iree/hal/cts/cts_test_base.h:93] Creating driver 'dylib'...
I ../iree/hal/cts/cts_test_base.h:66] Creating default device...
I ../iree/hal/cts/cts_test_base.h:68] Created device 'Dynamic Library (dylib)'
I ../iree/hal/cts/command_queue_test.cc:47] Device has 1 dispatch queue(s)
I ../iree/hal/cts/command_queue_test.cc:54] Device has 1 transfer queue(s)
[       OK ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/dylib (0 ms)
[ RUN      ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/llvm
I ../iree/hal/cts/cts_test_base.h:93] Creating driver 'llvm'...
I ../iree/hal/cts/cts_test_base.h:66] Creating default device...
I ../iree/hal/cts/cts_test_base.h:68] Created device 'llvm'
I ../iree/hal/cts/command_queue_test.cc:47] Device has 1 dispatch queue(s)
I ../iree/hal/cts/command_queue_test.cc:54] Device has 1 transfer queue(s)
[       OK ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/llvm (2 ms)
[ RUN      ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/vmla
I ../iree/hal/cts/cts_test_base.h:93] Creating driver 'vmla'...
I ../iree/hal/cts/cts_test_base.h:66] Creating default device...
I ../iree/hal/cts/cts_test_base.h:68] Created device 'vmla'
I ../iree/hal/cts/command_queue_test.cc:47] Device has 1 dispatch queue(s)
I ../iree/hal/cts/command_queue_test.cc:54] Device has 1 transfer queue(s)
[       OK ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/vmla (0 ms)
[ RUN      ] AllDrivers/CommandQueueTest.EnumerateDeviceQueues/vulkan
I ../iree/hal/cts/cts_test_base.h:93] Creating driver 'vulkan'...
W ../iree/hal/cts/cts_test_base.h:60] Skipping test as driver is unavailable: UNAVAILABLE: Unable to open dynamic library:'libvulkan.so.1: cannot open shared object file: No such file or directory'
/tmpfs/src/github/iree/build_tools/cmake/run_test.sh: line 33: 20058 Segmentation fault      (core dumped) "$@"
[22:00:55 UTC] cleanup
[22:00:55 UTC] echo 'Cleaning up test environment'
Cleaning up test environment
[22:00:55 UTC] rm -rf /tmpfs/src/github/iree/build/iree_hal_cts_command_queue_test_test_tmpdir
```

Reply:
The code in the first commit _seems_ like it should work to me, since attempting to but failing to create a Vulkan driver is fine - all the other hal/cts/ tests do that. This particular test just happens to fail when using emulated timeline semaphores, so all of its test cases should be skipped when using the Vulkan driver.

I've refactored to filter in yet a different way and the CI appears happier now.